### ISSUE TITLE
When replacing a block with a single statement the statement is not visited

### ISSUE THREAD

**Issue by amasad** [babel/babel#2620]
Example: http://astexplorer.net/#/wjvUCsQhF4
input:
```
if (a) {
  while(a) 1;
}
```
output:
```
if (a) while (a) 1;
```
plugin:
```
export default function ({Plugin, types: t}) {
  return new Plugin('ast-transform', {
    visitor: {
      BlockStatement(node) {
        return node.body[0];
      },
      WhileStatement(node) {
        alert('here'); // never gets here
      }
    }
  });
}
```

**Comment by sebmck** [comment]
This is fixed in 6.0. @amasad I can look into fixing this in 5.x if you'd like? Or do you want to defer?

**Comment by amasad** [comment]
nah, that's fine. Going to give 6.0 a spin

**Comment by thejameskyle** [comment]
Fixed in 6.0 https://github.com/thejameskyle/__babel-bug-2620

**Comment by amasad** [comment]
ah thanks for checking
